app/models/user_group.rb

```diff
@@ -79,6 +79,14 @@ def owns?(resource)
                             resource_type: resource.class.to_s)
   end
 
+  def has_admin?(user)
+    membership_for_user(user).try { |m| m.roles.include?("group_admin") }
+  end
+
+  def membership_for_user(user)
+    memberships.find_by(user_id: user.id)
```

Comment:
Membership has a unique index on (user_id, user_group_id) so can only ever be one membership.